x-pack/plugins/maps/public/classes/sources/table_source/table_source.ts

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import uuid from 'uuid';
+import { FIELD_ORIGIN, SOURCE_TYPES, VECTOR_SHAPE_TYPE } from '../../../../common/constants';
+import {
+  MapExtent,
+  MapFilters,
+  MapQuery,
+  TableSourceDescriptor,
+  TableSourceValue,
+  VectorJoinSourceRequestMeta,
+  VectorSourceSyncMeta,
+} from '../../../../common/descriptor_types';
+import { Adapters } from '../../../../../../../src/plugins/inspector/common/adapters';
+import { ITermJoinSource } from '../term_join_source';
+import { BucketProperties, PropertiesMap } from '../../../../common/elasticsearch_util';
+import { IField } from '../../fields/field';
+import { Query } from '../../../../../../../src/plugins/data/common/query';
+import {
+  AbstractVectorSource,
+  BoundsFilters,
+  GeoJsonWithMeta,
+  IVectorSource,
+  SourceTooltipConfig,
+} from '../vector_source';
+import { DataRequest } from '../../util/data_request';
+import { TableField } from '../../fields/table_field';
+
+export class TableSource extends AbstractVectorSource implements ITermJoinSource, IVectorSource {
```

Comment:
Why does `TableSource` need to implement `IVectorSource`?

Reply:
This is because `IField#getSource(): IVectorSource`

This method is just there to push along indexpattern and related ES-fucntionality to some UX-components. Will use separate PR to clean this up.